### ISSUE TITLE
How to set user level?

### ISSUE THREAD

**Issue by embeh** [meeting-room-booking-system/mrbs-code#3785]
We are using LDAP in MRBS version 1.11.6 to grant our users access to the booking system. 
Is there a way to **set the user level** (in the code referred to as `$mrbs_user->level`) through config.inc.php?

For example, we could define a few "full admins" that get `$max_level` as user level, and all normal users would get some other value?

**Comment by campbell-m** [comment]
There are two ways you could do this.

The first is to define your auth type, eg 'my_ldap', and then just override the `getUserFresh()` method.  For example something like:

~~~php
<?php
namespace MRBS\Auth;

class AuthMyLdap extends AuthLdap
{
   protected function getUserFresh(string $username) : ?User
  {
    $user = parent::getUserFresh($username);
    if (isset($user))
    {
       // Change $user->level here depending on who the user is
    }
    return $user;
  }
}
~~~

The other thing you could do is switch to using the [roles branch](https://github.com/meeting-room-booking-system/mrbs-code/tree/roles) which lets you define roles and groups and then assign roles to users and groups.  Roles can be set up to give read, write or admin permissions by room and area.  But be warned: the branch is still under development and it will also upgrade your database in a way that is not compatible with 1.11.6.  So if you go down that route you might want to try it out with a test installation first.

**Comment by embeh** [comment]
Thanks for that! I had to add `use MRBS\User;` but otherwise this worked pretty well.

One small thing I noted: A user with userlevel==2 is allowed edit the system message...

**Comment by campbell-m** [comment]
> Some small things I noted:

I've now fixed these in [9a2cc2c](https://github.com/meeting-room-booking-system/mrbs-code/commit/9a2cc2c753ff38dcd7ff27a02839241a2c1d247d).  The only exception is for editing the system message where it's deliberate that booking admins can do this.  The use case scenario is for example an outdoor tennis club, where a low level administrator might need to be able to put up a message such as "Courts closed today due to snow".  If you want to restrict it to just top level admins then edit admin.php and change line 276 from:

~~~php
if (is_book_admin())
~~~

to

~~~php
if (is_admin())
~~~

and also edit `get_page_level()` in mrbs_auth.inc and delete the two lines:

~~~php
    case 'edit_message.php':            // Booking admins can edit messages
    case 'edit_message_handler.php':    // Booking admins can edit messages
~~~

**Comment by embeh** [comment]
Thanks a lot!
